### ISSUE TITLE
test_add_archive_content started to fail since git-annex 0804

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#6923]
```shell
(git)smaug:/mnt/datasets/datalad/ci/git-annex/builds/2022/08[master]git
$> git grep 'FAILED.*test_add_archive_content' | head -n 1
cron-20220804/build-macos.yaml-779-841b46fb-failed/1_test-datalad (master).txt:2022-08-04T05:38:39.9259280Z FAILED ../datalad/local/tests/test_add_archive_content.py::test_add_archive_content

```
that is of 10.20220724+git60-g2930afded 

```
2022-08-04T05:38:39.9197700Z         with chpwd(repo_path):
2022-08-04T05:38:39.9197950Z             # test with excludes and renames and annex options
2022-08-04T05:38:39.9198190Z             ds.add_archive_content(
2022-08-04T05:38:39.9198560Z                 '1.tar.gz', exclude=['d'], rename=['/ /_', '/^1/2'],
2022-08-04T05:38:39.9198950Z                 annex_options="-c annex.largefiles=exclude=*.txt",
2022-08-04T05:38:39.9199200Z                 delete=True)
2022-08-04T05:38:39.9199410Z             # no conflicts since new name
2022-08-04T05:38:39.9199780Z >           ok_file_under_git('2', '1_f.txt', annexed=False)
2022-08-04T05:38:39.9199940Z
2022-08-04T05:38:39.9200320Z ../../../../hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/local/tests/test_add_archive_content.py:313:
2022-08-04T05:38:39.9200660Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2022-08-04T05:38:39.9201160Z ../../../../hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/tests/utils_pytest.py:425: in ok_file_under_git
2022-08-04T05:38:39.9201550Z     assert_in(file_repo_path, repo.get_indexed_files())  # file is known to Git
2022-08-04T05:38:39.9201840Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2022-08-04T05:38:39.9201970Z
2022-08-04T05:38:39.9202120Z first = '2/1_f.txt'
2022-08-04T05:38:39.9202770Z second = ['.datalad/.gitattributes', '.datalad/config', '.gitattributes', '1/1 f-1.1.txt', '1/1 f-1.2.txt', '1/1 f-1.txt', ...]
2022-08-04T05:38:39.9203100Z msg = None
2022-08-04T05:38:39.9203220Z
2022-08-04T05:38:39.9203320Z     def assert_in(first, second, msg=None):
2022-08-04T05:38:39.9203690Z         if msg is None:
2022-08-04T05:38:39.9203890Z >           assert first in second
2022-08-04T05:38:39.9204470Z E           AssertionError: assert '2/1_f.txt' in ['.datalad/.gitattributes', '.datalad/config', '.gitattributes', '1/1 f-1.1.txt', '1/1 f-1.2.txt', '1/1 f-1.txt', ...]
2022-08-04T05:38:39.9204730Z
2022-08-04T05:38:39.9205180Z ../../../../hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/tests/utils_pytest.py:131: AssertionError
```

**Comment by yarikoptic** [comment]
most likely the git/git-annex regression is between those two versions
```
(git)smaug:/mnt/datasets/datalad/ci/git-annex/builds/2022/08[master]git
$> grep '10\.2022' cron-20220803/build-ubuntu.yaml-*/1_test-datalad\ \(master\).txt | head -n 1         
2022-08-03T03:32:18.4144715Z   name: git-annex-debianstandalone-packages_10.20220724+git48-gd297e9ba7_amd64

$> grep '10\.2022' cron-20220804/build-ubuntu.yaml-783-841b46fb-failed/1_test-datalad\ \(master\).txt | head -n 1
2022-08-04T03:26:22.2612130Z   name: git-annex-debianstandalone-packages_10.20220724+git60-g2930afded_amd64
```

**Comment by yarikoptic** [comment]
meanwhile reported what I know to @joeyh at https://git-annex.branchable.com/bugs/add_--dry-run_seems_have_introduced_a___34__regression__34__/?updated

**Comment by yarikoptic** [comment]
fixed in [10.20220724-96-g21cfd0ea9](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=21cfd0ea980feaec8875380707c1a1ddb91de5dc)
